packages/wallets/src/features/cashier/flows/WalletWithdrawal/__tests__/WalletWithdrawal.spec.tsx

```diff
@@ -0,0 +1,141 @@
+        const replaceStateSpy = jest.spyOn(window.history, 'replaceState');
+        mockUseActiveWalletAccount.mockReturnValue({
+            // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+            data: {
+                currency: 'USD',
+            },
+        });
+
+        // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+        mockUseCurrencyConfig.mockReturnValue({
+            getConfig: jest.fn(),
+            isSuccess: true,
+        });
+
+        render(<WalletWithdrawal />);
+
+        expect(replaceStateSpy).toBeCalledWith({}, '', 'http://localhost/redirect');
+    });
+
+    it('should render withdrawal email verification page if no verification code found', () => {
+        Object.defineProperty(window, 'location', {
+            value: new URL('http://localhost/redirect'),
+            writable: true,
+        });
+
+        mockUseActiveWalletAccount.mockReturnValue({
+            // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+            data: {
+                currency: 'USD',
+            },
+        });
+
+        // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+        mockUseCurrencyConfig.mockReturnValue({
+            getConfig: jest.fn(),
+            isSuccess: true,
+        });
+
+        render(<WalletWithdrawal />);
+        expect(screen.getByText('WithdrawalVerificationModule')).toBeInTheDocument();
+    });
+
+    it('should render withdrawal fiat module if withdrawal is for fiat wallet', () => {
+        mockUseActiveWalletAccount.mockReturnValue({
+            // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+            data: {
+                currency: 'USD',
+            },
+        });
+
+        mockUseCurrencyConfig.mockReturnValue({
+            // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+            getConfig: jest.fn(() => ({ is_fiat: true })),
+            isSuccess: true,
+        });
+
+        render(<WalletWithdrawal />);
+        expect(screen.getByText('WithdrawalFiatModule')).toBeInTheDocument();
+    });
+
+    it('should render withdrawal crypto module if withdrawal is for crypto wallet', () => {
+        mockUseActiveWalletAccount.mockReturnValue({
+            // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+            data: {
+                currency: 'BTC',
+            },
+        });
+
+        mockUseCurrencyConfig.mockReturnValue({
+            // @ts-expect-error - since this is a mock, we only need partial properties of the hook
+            getConfig: jest.fn(() => ({ is_fiat: false })),
+            isSuccess: true,
+        });
+
+        render(<WalletWithdrawal />);
+        expect(screen.getByText('WithdrawalCryptoModule')).toBeInTheDocument();
```

Comment:
potentially worth testing that component is not only rendered, but also receives verificationCode (as without that, the functionality will be broken completely)

Reply:
same for WithdrawalFiatModule

Reply:
Good catch, will do this 👍